Added Chrony support

salvatore-coppola opened the pull request:
First try of implementation of nts via chrony.

salvatore-coppola committed:
First implementation.

salvatore-coppola committed:
Several Bugfixes

MMaiero commented on kura/org.eclipse.kura.linux.clock/OSGI-INF/metatype/org.eclipse.kura.clock.ClockService.xml:
```diff
@@ -43,8 +43,10 @@
+            
+               <Option label="java-ntp" value="java-ntp"/>
+               <Option label="ntpd"     value="ntpd"/>
+               <Option label="nts"      value="nts"/>
```
Can we label this "chrony-avanced"?

MMaiero commented on kura/org.eclipse.kura.linux.clock/OSGI-INF/metatype/org.eclipse.kura.clock.ClockService.xml:
```diff
@@ -107,6 +109,20 @@
             default="/dev/rtc0" 
             description="The RTC File Name. It defaults to /dev/rtc0."/>
+            
+        <AD id="clock.nts.config"
```
Can we name it "chrony.advanced.config"?

MMaiero commented on kura/org.eclipse.kura.linux.clock/OSGI-INF/metatype/org.eclipse.kura.clock.ClockService.xml:
```diff
@@ -107,6 +109,20 @@
+            required="false"
+            description="chrony configuration file.|TextArea" />
+
+		<AD id="clock.nts.config.location" 
```
Can we name it "chrony.advanced.location"?
Is this needed or we have a default chrony location?
If it can be changed, I believe we need to define a default location.
If it cannot be changed, maybe this field is not needed.

salvatore-coppola replied:
I have noticed that the dafult location could be either /etc/chrony.conf or /etc/chrony/chrony/conf so I have added it as input. Maybe i can check che existence of the configuration file via code

salvatore-coppola replied:
ok i'm going to remove the attribute ad the provider will try to find the right location

MMaiero commented on kura/test/org.eclipse.kura.linux.clock.test/src/test/java/org/eclipse/kura/linux/clock/NtsClockSyncProviderTest.java:
```diff
@@ -0,0 +1,15 @@
+package org.eclipse.kura.linux.clock;
```
Missing copyright

salvatore-coppola committed:
Bugfixes and Test added.

salvatore-coppola committed:
Chrony conf updated.

nicolatimeus commented:
I think we should also update `distrib/config/kura.build.properties` to have the new version included in the distributions

nicolatimeus commented on kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/NtsClockSyncProvider.java:
```diff
@@ -0,0 +1,254 @@
+        if (this.ntsConfig != null && !this.ntsConfig.isEmpty()) {
+            try {
+                logger.info("Saving previous chrony configuration file at {}", this.chronyConfigLocation);
+                Files.copy(this.chronyConfigLocation, Paths.get(this.chronyConfigLocation.toString() + ".bak"));
```
This seems to fail if the destination file exists, we should also check what happens if the source file does not exist

nicolatimeus commented on kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/NtsClockSyncProvider.java:
```diff
@@ -0,0 +1,254 @@
+        // check either chronyd or chrony unit because both are used in journal alternatively
+        Command journalClockUpdateRead = new Command(
+                new String[] { "journalctl", "-r", "-n", "1", "-u", NTS_PROVIDER, "-u", "chrony", "-b", "-o", "json",
+                        "-S", "today", "--output-fields", "MESSAGE", "-g", "'System clock was stepped by'" });
```
journalctl pattern matching might not be supported on all platforms.

salvatore-coppola committed:
Added missing copyright.
Added check on file existence.
Removed nts.config.location attribute. Now we check default locations.

nicolatimeus commented on kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/NtsClockSyncProvider.java:
```diff
@@ -0,0 +1,268 @@
+            }
+
+            if (chronyConfigLocation == null)
+                throw new KuraException(KuraErrorCode.CONFIGURATION_ATTRIBUTE_INVALID);
```
checkstyle seems to complain about missing braces

salvatore-coppola committed:
Fixed style problem.

salvatore-coppola committed:
Removed keyfile from config example.
Now lastSync is gathered from the journal entry.

salvatore-coppola committed:
Replaced inner grep in journalctl with external grep.

salvatore-coppola committed:
Updated org.eclipse.linu.clock bundle version to 1.1.0.
Added chrony as optional dependency to builds.
Updated the clock provider message.

salvatore-coppola committed:
Added SNAPSHOT qualifier to version.

salvatore-coppola committed:
Fixed version error.

MMaiero commented on kura/distrib/pom.xml:
```diff
@@ -924,7 +924,7 @@
                     <groupId>de.dentrassi.maven</groupId>
                     <artifactId>rpm</artifactId>
-                    <version>0.10.2</version>
+                    <version>1.5.0</version>
```
This needs a CQ

MMaiero commented on kura/org.eclipse.kura.linux.clock/OSGI-INF/metatype/org.eclipse.kura.clock.ClockService.xml:
```diff
@@ -42,9 +42,11 @@
+            
+               <Option label="java-ntp" value="java-ntp"/>
+               <Option label="ntpd"     value="ntpd"/>
+               <Option label="chrony-avanced"	value="chrony-avanced"/>
```
missing d in advanced

salvatore-coppola committed:
Added a content configuration check to avoid useless rewrite.
Fixed typo in chrony-provider name.

MMaiero commented:
@salvatore-coppola Is this ready for review?

salvatore-coppola commented:
yes

salvatore-coppola commented:
> 
> 
> @salvatore-coppola Is this ready for review?

yes

salvatore-coppola committed:
Nts classes now are Chrony classes.
Now chrony is a mandatory dependency for all env profiles.

MMaiero reviewed (commented):
I have tested on my raspberry pi and let it run for a while.
I have noticed that the code is a little bit too chatty, in my opinion.
Here a sample:
```
2021-07-14T05:42:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:42:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:43:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:43:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:44:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:44:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:45:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:45:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:46:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:46:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:47:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:47:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:48:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:48:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:49:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:49:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:50:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:50:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:51:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:51:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:52:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:52:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:53:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:53:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:54:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:54:13,436 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:55:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:55:13,436 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:56:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:56:13,437 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:56:44,136 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T05:56:44,137 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T05:56:44,153 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T05:57:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:57:13,461 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:58:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:58:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T05:59:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T05:59:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:00:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:00:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:01:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:01:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:02:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:02:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:03:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:03:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:04:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:04:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:05:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:05:13,454 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:06:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:06:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:07:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:07:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:08:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:08:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:09:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:09:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:10:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:10:13,461 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:11:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:11:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:11:44,154 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T06:11:44,154 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T06:11:44,171 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T06:12:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:12:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:13:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:13:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:14:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:14:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:15:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:15:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:16:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:16:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:17:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:17:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:18:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:18:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:19:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:19:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:20:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:20:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:21:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:21:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:22:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:22:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:23:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:23:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:24:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:24:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:25:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:25:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:26:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:26:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:26:44,172 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T06:26:44,173 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T06:26:44,189 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T06:27:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:27:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:28:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:28:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:29:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:29:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:30:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:30:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:31:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:31:13,465 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:32:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:32:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:33:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:33:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:34:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:34:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:35:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:35:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:36:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:36:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:37:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:37:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:38:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:38:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:39:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:39:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:40:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:40:13,456 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:41:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:41:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:41:44,190 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T06:41:44,191 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T06:41:44,207 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T06:42:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:42:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:43:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:43:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:44:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:44:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:45:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:45:13,457 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:46:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:46:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:47:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:47:13,457 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:48:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:48:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:49:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:49:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:50:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:50:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:51:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:51:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:52:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:52:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:53:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:53:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:54:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:54:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:55:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:55:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:56:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:56:13,454 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:56:44,209 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T06:56:44,209 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T06:56:44,227 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T06:57:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:57:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:58:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:58:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T06:59:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T06:59:13,458 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:00:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:00:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:01:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:01:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:02:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:02:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:03:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:03:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:04:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:04:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:05:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:05:13,454 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:06:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:06:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:07:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:07:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:08:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:08:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:09:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:09:13,459 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:10:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:10:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:11:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:11:13,457 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
2021-07-14T07:11:44,228 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask started.
2021-07-14T07:11:44,228 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask: Delete confirmed messages...
2021-07-14T07:11:44,245 [HouseKeeperTask] INFO  o.e.k.c.d.s.HouseKeeperTask - HouseKeeperTask ended.
2021-07-14T07:12:13,381 [pool-14-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Starting read the journal for clock updates...
2021-07-14T07:12:13,460 [pool-14-thread-1] WARN  o.e.k.l.c.ChronyClockSyncProvider - Unable to get system clock status from system journal.
```

Maybe we can keep those messages in debug, instead of info, just to be able to use them when something goes wrong, but for normal usage, I think it will fill up the logs. What do you think?

salvatore-coppola committed:
Now only clock stepping is logged with info level.

salvatore-coppola commented:
@MMaiero i have changed the logger levels. Now only the clock stepping event is logged with info level.

MMaiero commented:
I was checking the PR with the latest changes.
I am still getting some messages, are them correct?

```
2021-07-14T11:15:04,986 [ConfigurationListener Event Queue] INFO  o.e.k.l.c.ChronyClockSyncProvider - Clock synced
2021-07-14T11:15:05,051 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
2021-07-14T11:16:05,071 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
2021-07-14T11:17:05,074 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
```

salvatore-coppola committed:
Now logging of clock stepping is done only for new events.

salvatore-coppola commented:
> 
> 
> I was checking the PR with the latest changes.
> I am still getting some messages, are them correct?
> 
> ```
> 2021-07-14T11:15:04,986 [ConfigurationListener Event Queue] INFO  o.e.k.l.c.ChronyClockSyncProvider - Clock synced
> 2021-07-14T11:15:05,051 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
> 2021-07-14T11:16:05,071 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
> 2021-07-14T11:17:05,074 [pool-32-thread-1] INFO  o.e.k.l.c.ChronyClockSyncProvider - Journal successfully readed. Last clock stepping event was at: 2021-07-14T10:15:04.806545Z
> ```

There should be only one log line, I committed a change to be sure of that.

nicolatimeus commented on kura/org.eclipse.kura.linux.clock/src/main/java/org/eclipse/kura/linux/clock/ChronyClockSyncProvider.java:
```diff
@@ -0,0 +1,293 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
```
I think this should be `org.slf4j.Logger`

salvatore-coppola replied:
yes

salvatore-coppola committed:
Replaced wrong logging package.

nicolatimeus commented:
There seem to be ClockService related test failures on Jenkins

salvatore-coppola commented:
> There seem to be ClockService related test failures on Jenkins

uhm, the same test run fine on my machine:
![immagine](https://user-images.githubusercontent.com/83589980/125642788-e452eeff-1ef3-4659-95b8-2c65696f0fd8.png)

salvatore-coppola commented:
@MMaiero have you seen 

> 
> 
> The backport to `release-5.0.0` failed:
> 
> ```
> The process '/usr/bin/git' failed with exit code 1
> ```
> 
> To backport manually, run these commands in your terminal:
> 
> ```shell
> # Fetch latest updates from GitHub
> git fetch
> # Create a new working tree
> git worktree add .worktrees/backport-release-5.0.0 release-5.0.0
> # Navigate to the new working tree
> cd .worktrees/backport-release-5.0.0
> # Create a new branch
> git switch --create backport-3471-to-release-5.0.0
> # Cherry-pick the merged commit of this pull request and resolve the conflicts
> git cherry-pick --mainline 1 6b5b683ab484e0310c783bb62cdc82667c8fbca5
> # Push it to GitHub
> git push --set-upstream origin backport-3471-to-release-5.0.0
> # Go back to the original working tree
> cd ../..
> # Delete the working tree
> git worktree remove .worktrees/backport-release-5.0.0
> ```
> 
> Then, create a pull request where the `base` branch is `release-5.0.0` and the `compare`/`head` branch is `backport-3471-to-release-5.0.0`.

@MMaiero have you seen this ?

MMaiero commented:
Yes, I should have created a back port PR
